src/Controller/SalesChannelApiController.php

```diff
@@ -144,14 +153,15 @@ public function postPaymentDetails(
      * @Route(
      *     "/sales-channel-api/v1/adyen/payment-status",
      *     name="sales-channel-api.action.adyen.payment-status",
-     *     methods={"GET"}
+     *     methods={"POST"}
      * )
      *
+     * @param Request $request
      * @param SalesChannelContext $context
      * @return JsonResponse
      */
-    public function getPaymentStatus(SalesChannelContext $context): JsonResponse
+    public function getPaymentStatus(Request $request): JsonResponse
     {
-        return new JsonResponse(['aloha']);
+        return new JsonResponse($this->paymentStatusService->getPaymentStatusWithOrderId($request->get('orderId')));
```

Comment:
you could still use the function I created with the SalesChannelContext token and the order id - is there a reason why you only filter by order id?

Reply:
Done!